schematic/utils/general.py

```diff
@@ -114,7 +116,21 @@ def entity_type_mapping(syn, entity_id):
     else:
         # if there's no matching type, return concreteType
         return entity.concreteType
-        
+
+def create_temp_folder(path: str) -> str:
+    """This function creates a temporary directory in the specified directory 
+    Args:
+        path(str): a directory path where all the temporary files will live
+    Returns: returns the absolute pathname of the new directory.
+    """
+    # Create a temporary directory in the specified directory
+    try:
+        path = tempfile.mkdtemp(dir=path)
+        return path
+    except FileNotFoundError:
```

Comment:
Why is the same exception being raised after being caught?
If that is intended, I would [chain](https://docs.python.org/3/tutorial/errors.html#exception-chaining) this exception.

Reply:
Thanks for the feedbacks @andrewelamb and @thomasyu888 . I will update the code to remove try except.